scbctl/cmd/scans.go

```diff
@@ -116,6 +126,74 @@ func NewScanCommand() *cobra.Command {
 	}
 
 	scanCmd.Flags().String("name", "", "Name of the created scan. If no name is provided, the ScanType will be used as the name")
+	scanCmd.Flags().Bool("follow", false, "Follow the log output")
 
 	return scanCmd
 }
+
+func followScanLogs(ctx context.Context, kubeclient client.Client, namespace, scanName string) error {
+	// Find the job associated with the scan
+	jobList := &batchv1.JobList{}
+	labelSelector := client.MatchingLabels{}
+
+	fmt.Println("Listing jobs in namespace:", namespace)
+
+	for {
+		fmt.Println("Attempting to list jobs...")
+		err := kubeclient.List(ctx, jobList, client.InNamespace(namespace), labelSelector)
+		if err != nil {
+			return fmt.Errorf("error listing jobs: %s", err)
+		}
+
+		if len(jobList.Items) == 0 {
+			fmt.Println("No jobs found, retrying...")
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		fmt.Printf("Found %d job(s)\n", len(jobList.Items))
+
+		for _, j := range jobList.Items {
+			fmt.Printf("Job: %s, Labels: %v\n", j.Name, j.Labels)
+		}
+
+		var job *batchv1.Job
+		for _, j := range jobList.Items {
+			fmt.Printf(j.Name)
+			if strings.HasPrefix(j.Name, fmt.Sprintf("scan-%s", scanName)) {
+				job = &j
+				break
+			}
+		}
+
+		if job == nil {
+			fmt.Println("No matching job found, retrying...")
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		jobName := job.Name
+		containerName := scanName
+
```

Comment:
The scanName isn't equal to the container name. At least not in all cases.
The "primary" container for which the logs are the most interessting, and which the SCB considers to be the deciding factor if the scan is finsihed is always the first one, so it should be

```suggestion
		containerName := job.Spec.Template.Spec.Containers[0].Name
```